zetta_utils/layer/volumetric/cloudvol/backend.py

```diff
@@ -299,6 +306,12 @@ def write(self, idx: VolumetricIndex, data: torch.Tensor):
         cvol[slices] = data_final
         cvol.autocrop = False
 
+    def as_type(self, backend_type) -> VolumetricBackend:  # pragma: no cover # type: ignore
+        type_args = [f.name for f in attrs.fields(backend_type)]
+        keys_to_use = set(dir(self)).intersection(type_args)
+        kwargs_to_use = {k: getattr(self, k) for k in keys_to_use}
+        return backend_type(**kwargs_to_use)
```

Comment:
Why don't we just have a function `to_tensorstore_backend` that takes a volumetric backend and returns `tensorstore` backend? Using inheritance here leads to ugly typing.

Also, does the following not work here?
```python
from typing import Type

T = TypeVar('T', bound=VolumetricBackend)

def as_type(self, backend_type: Type[T]) -> T:
   ...
```

Reply:
The reason was that I wanted to avoid a circular import and didn't want to hardcode `if check_type(layer.backend, CVBackend) or check_type(layer.backend, VolumetricLayerSetBackend): layer.backend.to_tensorstore_backend` in volumetric_apply_flow. I'll factor out the PrecomputedInfoSpec (alongside the `__attrs_post_init__` below), and just add it as a hardcode (rather than a file-level variable).

The typing should work if I'm not inheriting.

Reply:
Hmm, I think that the typing cannot actually be fixed because of the `VolumetricSetBackend` unless we want to make a VolumetricSetBackend be a subtype of TSBackend. I'll leave `to_tensorstore_backend` as something that returns `VolumetricBackend` even though that's not perfect.